common/v2/features/SignAndVerifyMessage/SignMessage.tsx

```diff
@@ -0,0 +1,237 @@
+        throw Error;
+      }
+
+      const address = wallet.getAddressString();
+      let sig = '';
+
+      if (walletName === SecureWalletName.PARITY_SIGNER) {
+        const data = messageToData(message);
+        props.requestMessageSignature(address, data);
+      } else {
+        sig = await wallet.signMessage(message, props.nodeLib);
+      }
+
+      const combined = {
+        address,
+        msg: message,
+        sig,
+        version: '2'
+      };
+      setError(undefined);
+      setSignedMessage(combined);
+      setIsSigned(!!sig);
+    } catch (err) {
+      setError(translateRaw('ERROR_38'));
+      setSignedMessage(null);
+    }
+  };
+
+  const handleOnChange = (msg: string) => {
+    setMessage(msg);
+    setError(undefined);
+    setSignedMessage(null);
+  };
+
+  const onSelect = (selectedWalletName: WalletName) => {
+    setWalletName(selectedWalletName);
+    setShowSubtitle(false);
+  };
+
+  const onUnlock = (selectedWallet: any) => {
+    setWallet(selectedWallet);
+    setUnlocked(true);
+  };
+
+  const resetWalletSelectionAndForm = () => {
+    setMessage('');
+    setError(undefined);
+    setSignedMessage(null);
+    setWalletName(undefined);
+    setUnlocked(false);
+    setIsSigned(false);
+    setShowSubtitle(true);
+  };
+
+  const story = STORIES.find(x => x.name === walletName);
+  const Step = story && story.steps[0];
+
+  return (
+    <Content>
+      {walletName ? (
+        <>
+          <BackButton marginBottom={unlocked} basic={true} onClick={resetWalletSelectionAndForm}>
+            <img src={backArrowIcon} alt="Back arrow" />
+            {translateRaw('CHANGE_WALLET_BUTTON')}
+          </BackButton>
+          {!unlocked && Step && (
+            <Step wallet={WALLET_INFO[walletName]} onUnlock={onUnlock} formData={defaultFormData} />
+          )}
+        </>
+      ) : (
+        <WalletList wallets={STORIES} onSelect={onSelect} />
+      )}
+
+      {unlocked && walletName && (
+        <>
+          <InputField
+            value={message}
+            label={translate('MSG_MESSAGE')}
+            placeholder={translateRaw('SIGN_MSG_PLACEHOLDER')}
+            textarea={true}
+            onChange={event => handleOnChange(event.target.value)}
+            height="150px"
+            inputError={error}
+          />
+          <SignButton disabled={!message} onClick={handleSignMessage}>
+            {translate('NAV_SIGNMSG')}
+          </SignButton>
+          {signedMessage && isSigned && (
+            <SignedMessage>
+              <SignedMessageLabel>{translate('MSG_SIGNATURE')}</SignedMessageLabel>
+              <CodeBlockWrapper>
+                <CodeBlock>{JSON.stringify(signedMessage, null, 2)}</CodeBlock>
+              </CodeBlockWrapper>
+            </SignedMessage>
+          )}
+        </>
+      )}
+    </Content>
+  );
+}
+
+const mapStateToProps = (state: AppState) => ({
+  nodeLib: configNodesSelectors.getNodeLib(state),
+  sig: state.paritySigner.sig
```

Comment:
Is it intentional to default to Parity?!

Reply:
This is parity-specific implementation, we renamed the variables to make that more clear.